tracing-attributes/src/lib.rs

```diff
@@ -468,8 +468,7 @@ fn gen_body(
         quote_spanned!(block.span()=>
             let __tracing_attr_span = #span;
             let __tracing_attr_guard = __tracing_attr_span.enter();
-            let f = move || #return_type { #block };
-            match f() {
+            match (move || #return_type { #block })() {
```

Comment:
hahaha cool it turns out there's a clippy lint against doing this specifically: https://github.com/tokio-rs/tracing/pull/1167/checks?check_run_id=1818724510#step:4:768

we could factor out the closure into an additional let-binding, like it was previously, but then we'd need to make it a `let mut`, and the `mut` _might_ not be used...so then we'd need an `#[allow(unused_mut)]` attribute. alternatively we could just add the clippy `#[allow(clippy::redundant_closure_call)]` attribute here, to tell Clippy to ignore this, which I think is probably the better option:
```suggestion
            #[allow(clippy::redundant_closure_call)]
            match (move || #return_type { #block })() {
```

Reply:
ah! i figured out what the problem is here! the `{` and `}` around `#block` are unnecessary --- the block token already includes them. so this should just be
```suggestion
            match (move || #return_type #block)() {
```
and then clippy will _actually_ be satisfied!